gen/bind_generator.jl

```diff
@@ -241,7 +241,13 @@ function _bind(__module__, __source__, sig::Expr, err::Union{String,Expr,Nothing
     # avoids inserting the line number nodes for the macro --- the call site
     # is instead explicitly injected into the function body via __source__.
     jlfuncsig = Expr(:call, jlfuncname, args...)
-    jlfuncbody = Expr(:block, __source__, :($statsym = $ccallexpr))
+    jlfuncbody = Expr(
+        :block, __source__, :(lock(liblock)), :($statsym = try
+            $ccallexpr
+        finally
+            unlock(liblock)
+        end)
+    )
```

Comment:
```suggestion
    jlfuncbody = Expr(
        :block, __source__, :(use_lock() && lock(liblock)), :($statsym = try
            $ccallexpr
        finally
            use_lock && unlock(liblock)
        end)
    )
```

Reply:
This adds a runtime lookup

Reply:
If `use_lock` is constant, (e.g. `use_lock() = false`), it gets inlined. The generated code does not actually call `use_lock` at runtime.

Reply:
Ah, I saw that's what you did in #1022. That's an option, but I would argue that using the lock should be the default.